src/v/cluster/data_migration_table.cc

```diff
@@ -201,6 +201,14 @@ migrations_table::apply(create_data_migration_cmd cmd) {
 std::optional<migrations_table::validation_error>
 migrations_table::validate_migrated_resources(
   const data_migration& migration) const {
+    // cloud_storage_api is checked on startup
+    if (!config::shard_local_cfg().cloud_storage_enabled()) {
```

Comment:
can we pass a configuration::binding for those two values ?, the code will be much easier to test

Reply:
it won't work, as these settings fail `base_property::assert_live_settable()`

Reply:
used a boolean as discussed